flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -397,8 +394,17 @@ public void setNumBuffers(int numBuffers) throws IOException {
 			returnExcessMemorySegments();
 
 			numExcessBuffers = numberOfRequestedMemorySegments - currentPoolSize;
+
+			if (currentPoolSize > numberOfRequestedMemorySegments
+					&& availableMemorySegments.isEmpty()
+					&& isAvailable(networkBufferPool.isAvailable())) {
+				toNotify = availabilityHelper.getUnavailableToResetUnavailable();
+			}
 		}
 
+		// notify if the pool size has been expanded
```

Comment:
This comment should be added exactly in above `if` condition and described more accurately if needed.